api/v1/services/MailService.js

```diff
@@ -0,0 +1,223 @@
+ * @param  {String|Array} recipients	a list of the recipients
+ * @param  {String} template			the template identifier to use
+ * @param  {Object} substitutions		a mapping of keys to their substitution values (optional)
+ * @return {Promise}					an empty promise
+ * @throws {ExternalProviderError} 		when the mail client returns an error
+ */
+function send(recipients, template, substitutions) {
+	var transmission = {
+		content: { template_id: template },
+		recipients: _formatRecipients(recipients, config.isDevelopment),
+		substitution_data: (substitutions) ? substitutions : {}
+	};
+
+	if (_.isEmpty(transmission.recipients)) {
+		logDisabled(template, recipients, substitutions, RECIPIENTS_NOT_WHITELISTED_REASON);
+		return Promise.resolve(true);
+	}
+
+	return transmissionsAsync
+		.sendAsync({ transmissionBody: transmission })
+		.then(function (result) {
+			// get rid of the transmission response
+			return true;
+		})
+		.catch(_handleClientError);
+}
+
+/**
+ * Sends a templated email to the requested email list
+ * @param  {Object} list          	the (internal-client) list representation (see mail.js)
+ * @param  {String} template      	the template identifier to use
+ * @param  {Object} substitutions 	a mapping of keys to their substitution values (optional)
+ * @return {Promise}				an empty promise
+ * @throws {ExternalProviderError}	when the mail client returns an error
+ */
+function sendToList(list, template, substitutions) {
+	if (config.isDevelopment && !_isWhitelistedList(list.name)) {
+		logDisabled(template, list.name, substitutions, LIST_NOT_WHITELISTED_REASON);
+		return Promise.resolve(true);
+	}
+
+	var recipientList = {
+		id: list.id,
+		recipients: [] // unknown until we gather the recipients from the datastore
+	};
+	var transmission = {
+		content: { template_id: template },
+		recipients: { list_id: list.id },
+		substitution_data: (substitutions) ? substitutions : {}
+	};
+	var isViableList = true;
+
+	return _formatRecipientsList(list.name)
+		.then(function (recipients) {
+			recipientList.recipients = recipients;
+			if (recipients.length >= 1) {
+				// we cannot update a list with any less than 1 member
+				return recipientsAsync.updateAsync(recipientList);
+			}
+
+			return true;
+		})
+		.then(function () {
+			if (recipientList.recipients.length === 0) {
+				logDisabled(template, list.name, substitutions, "the recipient list was empty");
+				return true;
+			}
+			if (recipientList.recipients.length === 1) {
+				logDisabled(template, list.name, substitutions, "the recipient list held only" +
+					" one recipient; sending via regular transmission");
+
+				var recipient = recipientList.recipients[0].address.email;
+				return send(recipient, template, substitutions);
+			}
+			return transmissionsAsync.sendAsync({ transmissionBody: transmission });
+		})
+		.then(function () {
+			// get rid of the transmission response, if there is one
+			return true;
+		})
+		.catch(_handleClientError);
+}
+
```

Comment:
Would it not make sense to have the below 2 methods addToList and removeFromList defined in the MailingList.js as member functions and wrap that here since they pretty clearly operate like a member function and it would be easier to reuse